.github/actions/load-workflow-variables/action.yml

```diff
@@ -18,93 +18,92 @@ description: |-
+            echo "ERROR: ${FULL_WORKING_DIR} is not a sub-directory of ${GITHUB_WORKSPACE}"
+            exit 1
+        fi
 
-        if [[ ! -f "${SEARCH_PATH}" ]]; then
-            echo "ERROR: Failed to find target file ${SEARCH_PATH}"
+        FILEPATH=$(realpath "${FILEPATH}")
+        if [[ "${FILEPATH}" != "${GITHUB_WORKSPACE}"* ]]; then
+            echo "ERROR: ${FILEPATH} is not a sub-directory of ${GITHUB_WORKSPACE}"
             exit 1
         fi
 
-        PROPS=$(yq -o=props '.. | select(tag != "!!map" and tag != "!!seq") | ( (path | join("_")) + "<<EOT\n" + . + "\nEOT")' "${SEARCH_PATH}")
+        WORKING_DIRECTORY=$(echo "${FULL_WORKING_DIR}" | sed -r "s|^${GITHUB_WORKSPACE}\/||g")
 
-        for PROP in "${PROPS}"; do
-            echo "${PROP}" >> $GITHUB_ENV
-        done
+        echo "FULL_WORKING_DIR:${FULL_WORKING_DIR}"
+        echo "WORKING_DIRECTORY:${WORKING_DIRECTORY}"
+        echo "FILEPATH:${FILEPATH}"
 
-    - name: 'Load From Local Or Parents'
-      if: |-
-        contains(fromJSON('["true", "True", "TRUE", "1", "T", "t"]'), inputs.find_in_parents)
-      shell: 'bash'
-      env:
-        WORKING_DIRECTORY: '${{ inputs.working_directory }}'
-        ENV_FILENAME: '${{ inputs.env_filename }}'
-        TOP_LEVEL_DIRECTORY: '${{ inputs.top_level_directory || github.workspace }}'
-      run: |-
-        WORKING_DIRECTORY=$(realpath "${WORKING_DIRECTORY}")
-        TOP_LEVEL_DIRECTORY=$(realpath "${TOP_LEVEL_DIRECTORY}")
+        # YAML provides a more human friendly way to define configs, but yq
+        # does prefix/suffix matching on * which can cause unwanted results.
+        # jq does exact matching, so we use yq to convert YAML to JSON, then
+        # use jq to do exact matching for directory values.
 
-        if [[ "${WORKING_DIRECTORY}" != "${TOP_LEVEL_DIRECTORY}"* ]]; then
-          echo "ERROR: ${TOP_LEVEL_DIRECTORY} is not a sub-directory of ${WORKING_DIRECTORY}"
-          exit 1
-        fi
+        # Convert YAML to JSON
+        JSON=$(yq -o=json "${FILEPATH}")
 
-        SEARCH_PATH="${WORKING_DIRECTORY}/${ENV_FILENAME}"
+        # Look for exact match on working directory
+        FOUND=$(echo "${JSON}" | jq -r ".[] | select(.dirs[] | select(. == \"$WORKING_DIRECTORY\"))")
 
-        echo -n "Checking ${SEARCH_PATH}..."
-        if [[ ! -f "${SEARCH_PATH}" ]]; then
-          echo "MISSING"
+        if [[ -z "${FOUND}" ]]; then
+            echo "NOT FOUND   No exact match for ${WORKING_DIRECTORY}"
 
-          # failsafe to prevent infinite loop
-          while [[ "${WORKING_DIRECTORY}" != "/"  ]]; do
-            WORKING_DIRECTORY=$(dirname "${WORKING_DIRECTORY}")
-            SEARCH_PATH="${WORKING_DIRECTORY}/${ENV_FILENAME}"
+            # Search directory hierarchy upwards for a * suffix indicating all children of parent be used
+            while [[ "${WORKING_DIRECTORY}" != "/" && "${WORKING_DIRECTORY}" != "." && "${WORKING_DIRECTORY}" != "" ]]; do
```

Comment:
This is pretty inefficient. What happens when the env file blows up (in the case of X's it will eventually get pretty large). It would probably be easier to just inline a python file and write that to tmp then execute it.